src/extension.ts

```diff
@@ -162,12 +180,48 @@ class RunOnSaveExtension {
       : vscode.workspace.rootPath;
   }
 
-  public get isEnabled(): boolean {
-    return !!this._context.globalState.get('isEnabled', true);
+  private _wakeSyncCommandRunner(): void {
+    const notifyWaitingRunner = this._notifyWaitingSyncCommandRunner;
+    if (notifyWaitingRunner) {
+      this._notifyWaitingSyncCommandRunner = undefined;
+      notifyWaitingRunner();
+    }
```

Comment:
I think this can be simplified:
```suggestion
    this._notifyWaitingSyncCommandRunner?.()
    this._notifyWaitingSyncCommandRunner = undefined;
```